### PR TITLE
fix tab title propagation issues

### LINKED ISSUE #8704: Title propagation issues (empty app title, title doesn't update when unfocused, etc.)
# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.19042.685]
Windows Terminal version (if applicable): latest on main branch but it also happens
on the preview version from the store
```

# Steps to reproduce

Open Windows terminal and <kbd>Alt</kbd>+<kbd>Tab</kbd>.

# Expected behavior

Shows Application title.

# Actual behavior

App title is empty.

# Context

These are my settings: https://gist.github.com/giggio/d4cf676d93b89a2f78cb7f25c01fdcda

You can see that there is no title in <kbd>Alt</kbd>+<kbd>Tab</kbd>:
image

It is also empty on the task bar:
image

And when using <kbd>Win</kbd>+<kbd>Tab</kbd>:
image

I'd like it to simply read "Windows Terminal", always.

### PR DESCRIPTION
**Pull request by sarim**
- Fixes empty app title when  `showTerminalTitleInTitlebar` is false
- Fixes Tab title propagation to Window title when
  `showTerminalTitleInTitlebar` is false
- Fixes Tab title propagation to Window - title doesn't update when
  Window is unfocused

1. There were a missing
   `_settings.GlobalSettings().ShowTitleInTitlebar()` check. Because of
   this Title update event was being fired even when
   `showTerminalTitleInTitlebar` is false. This results in empty tab
   title to propagate to Window title. Also then after switching tabs
   back and forth, tab title propagates to window title. These shouldn't
   propagate when `showTerminalTitleInTitlebar` is false. I added the
   `showTerminalTitleInTitlebar` check in relevant logic to fix the
   behavior. 

2. Code was checking `tab.FocusState() != FocusState::Unfocused` , but
   when the whole terminal window is not in focus, the active tab is
   also in Unfocused state. This was preventing tab title to propagate
   to window title when application is unfocused. I added the logic of
   checking matching selected tabs' index. This fixes the issue.

## Validation Steps Performed
I did the reproduce steps descripted in the issue to reproduce the bugs.
After applying the fixes, the bugs don't appear anymore while doing the
reproduce steps.

Closes #8704

### CODE REVIEW

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 1]
```diff
@@ -1065,8 +1065,13 @@ namespace winrt::TerminalApp::implementation
     {
         auto newTabTitle = tab.Title();
 
+        // When Application window is unfocused, selected tab's FocusState also
+        // becomes Unfocused. So we have to check using FocusedTabIndex too.
+        auto focusedTabIndex = _GetFocusedTabIndex();
+
         if (_settings.GlobalSettings().ShowTitleInTitlebar() &&
-            tab.FocusState() != FocusState::Unfocused)
+            (tab.FocusState() != FocusState::Unfocused ||
```
Comment (Don-Vito) [inline src/cascadia/TerminalApp/TerminalPage.cpp:1073]:
```
if (_settings.GlobalSettings().ShowTitleInTitlebar() && tab == _GetFocusedTab())
```
should do the job

Comment (sarim) [reply src/cascadia/TerminalApp/TerminalPage.cpp]: Done and pushed. btw `_GetFocusedTab()` can also return nullptr as per its description. Not sure its that can become a problem.

**Comment by Don-Vito** [comment]
@sarim - to fix the spell check please rebase/merge main.

**Comment by sarim** [comment]
@Don-Vito okey, I merged main.

**Review by carlos-zamora — APPROVED** [review]

**Review by Don-Vito — APPROVED** [review]

**Review by DHowett — APPROVED** [review]
This is a great and comprehensive first contribution. Thank you!

**Pull request merged**